app/Console/UpdateUniqueLabels.php

```diff
@@ -0,0 +1,143 @@
+<?php declare(strict_types=1);
+
+namespace CultuurNet\UDB3\Silex\Console;
+
+use CultuurNet\UDB3\Label\ValueObjects\LabelName;
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\DBALException;
+use Doctrine\DBAL\Exception\UniqueConstraintViolationException;
+use PDO;
+use Rhumsaa\Uuid\Uuid;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Helper\ProgressBar;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Question\ConfirmationQuestion;
+
+class UpdateUniqueLabels extends Command
+{
+    private const MAX_RESULTS = 1000;
+    private const LABEL_CREATED = 'CultuurNet.UDB3.Label.Events.Created';
+
+    /**
+     * @var Connection
+     */
+    private $connection;
+
+    public function __construct(Connection $connection)
+    {
+        parent::__construct();
+        $this->connection = $connection;
+    }
+
+    protected function configure(): void
+    {
+        $this->setName('label:update-unique')
+            ->setDescription('Updates the table with unique labels based on the `LabelAdded` events inside the event store.');
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $labelAddedEventsCount = $this->getAllLabelAddedEventsCount();
+
+        if ($labelAddedEventsCount <= 0) {
+            $output->writeln('No `LabelAdded` events found.');
+            return 0;
+        }
+
+        $helper = $this->getHelper('question');
+        $updateQuestion = new ConfirmationQuestion('Update ' . $labelAddedEventsCount . ' label(s)? [y/N] ', false);
+        if (!$helper->ask($input, $output, $updateQuestion)) {
+            return 0;
+        }
+
+        $progressBar = new ProgressBar($output, $labelAddedEventsCount);
+
+        $messages = [];
+        $offset = 0;
+        do {
+            $labelAddedEvents = $this->getAllLabelAddedEvents($offset);
+
+            foreach ($labelAddedEvents as $labelAddedEvent) {
+                $labelUuid = $this->getLabelUuid($labelAddedEvent);
+                $labelName = $this->getLabelName($labelAddedEvent);
+
+                try {
+                    $this->updateLabel($labelUuid, $labelName);
+                    $messages[] = 'Added label ' . $labelName->toNative() . ' with uuid ' . $labelUuid->toString();
+                } catch (UniqueConstraintViolationException $exception) {
+                    $messages[] = 'Unique exception for label ' . $labelName->toNative() . ' with uuid ' . $labelUuid->toString();
+                }
+
+                $progressBar->advance();
+            }
+
+            $offset += count($labelAddedEvents);
+        } while ($offset < $labelAddedEventsCount);
```

Comment:
It's been a long time since I've seen a do/while construct with a nested loop 😅

Reply:
Yes, a good old `do-while` construct! Lucky it's not a `for-each` inside another `for-each` 😅

I thought about:
- working with a generator but that would require a select statement for every single label created event
- creating a repository class around the event store

But this command should only be run once. So it's not worth the effort.